src/gluonts/dataset/common.py

```diff
@@ -26,6 +26,7 @@
     Union,
     cast,
 )
+from typing_extensions import Protocol
```

Comment:
This seems to be introduced in `3.8`.

There are some other nice things in `3.8`, which I would like to use. Maybe we can have a discussion if we want to require `3.8` at some point.

Reply:
Yes I thought the same. Also, the introduction of `Literal` is nice for multiple use cases.